core/module.go

```diff
@@ -15,6 +15,29 @@ import (
 	"github.com/dagger/dagger/engine/slog"
 )
 
+type SDKConfig struct {
+	Source string            `json:"source"`
+	Pin    string            `json:"pin"`
+	Env    map[string]string `json:"env"`
```

Comment:
To make these types queryable and appear in the schema, you'll need to add `field` tags, e.g. look at `ModuleSource`:

https://github.com/dagger/dagger/blob/539584e2515c32c2fb5fa053004c4c31e08ff0fa/core/modulesource.go#L58-L77

Reply:
with map[string]string we get `cannot convert map[string]string to a Typed value` error. how do you feel about making the env either `[]string` (meaning user will have to type in KEY=VALUE format) OR making it `[]Env`, where `Env` is a struct with a key/value fields.

Reply:
well apparently []Env also fails, so I have to keep it as []string. @jedevc what do you think from UX perspective here?

Reply:
Hm, having `[]Env` *should* work - we already have `BuildArg`, see: https://github.com/dagger/dagger/blob/12f298aad280ca957f3c69e9a589c48f7118e8b0/core/schema/query.go#L61-L61

That should allow you to have it in the API.

However, the `ModuleConfig` struct should ideally still represent it as a `map[string]string` (should be fine, since it's serialized as JSON, and I don't *think* it's exposed in the API directly).